### PR TITLE
Nginx directories creation and bigpipe support

### PR DESCRIPTION
**Pull request by ivrh**
1. Added creation of Nginx directories, 
2. added bigpipe support (via a variable)
3. Added 'auto' to the nginx worker processes to set it = to CPU count.

### CODE REVIEW

**Comment by FinBoWa** [comment]
@ivrh Why does have commits from 2015? I think you should rebase your commits top of the master

**Comment by ivrh** [comment]
Will review this, but forked about 2 weeks ago, not back in 2015

**Comment by FinBoWa** [comment]
Have you been merging changes from the upstream to your fork?

On a good pull request the old commits that are already in the repo shouldn't show up here at all only the commits that you introduce, so a rebasing your changes to the top of the latest master should clean up this nicely.

**Comment by FinBoWa** [comment]
As a suggest I'd do the following:

- create new branch on your fork from your master
- rebase geerlinguys master to that, and pick commits that you want and/squash some of them to a one to make the changes more clear > or just one if thats feasible 

Submit that as a new pull request.

Just to be clear here: I'm just commenting on the pull request, not will @geerlingguy  accept this pr :)

File: templates/headers.conf.j2 [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{% if nginx_bigpipe_enable %}
+  add_header X-Accel-Buffering: no;
+{% endif %}
+add_header X-Frame-Options SAMEORIGIN;
+add_header X-Content-Type-Options nosniff;
+add_header X-XSS-Protection "1; mode=block";
+add_header Content-Security-Policy "default-src 'self'; script-src 'self' 'unsafe-inline' 'unsafe-eval' https://ssl.google-analytics.com https://assets.zendesk.com https://connect.facebook.net; img-src 'self' https://ssl.google-analytics.com https://s-static.ak.facebook.com https://assets.zendesk.com; style-src 'self' 'unsafe-inline' https://fonts.googleapis.com https://assets.zendesk.com; font-src 'self' https://themes.googleusercontent.com; frame-src https://assets.zendesk.com https://www.facebook.com https://s-static.ak.facebook.com https://tautt.zendesk.com; object-src 'none'";
```
Comment (geerlingguy) [inline templates/headers.conf.j2]: There are a lot of extra changes in this PR that I'd rather not add full-stop, but rather discuss separately. Most of these changes would be better served in a per-server-block config, and not universally.

**Comment by geerlingguy** [comment]
Please rebase on master so the PR is a little easier to evaluate; also, it seems all the file modes were changed from 644 to 755.

I tend not to accept PRs that make more than one or two small/isolated changes, and it looks like this PR is making a number of changes that I might or might not want to include in the role.

**Pull request closed without merging**